test/with_dummyserver/test_connectionpool.py

```diff
@@ -1006,7 +1005,13 @@ def test_pool_size_retry(self):
 
 class TestRedirectPoolSize(HTTPDummyServerTestCase):
     def setUp(self):
-        self.pool = HTTPConnectionPool(self.host, self.port, maxsize=10, retries=5, block=True)
+        retries = Retry(
+            total=3,
+            raise_on_status=False,
+            status_forcelist=[404],
+        )
```

Comment:
This needs to set `redirect=True` to avoid breaking if we ever change the kwarg defaults.